engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -883,16 +936,18 @@ class Compiler(
     * @return the diagnostics from the module
     */
   def gatherDiagnostics(module: Module): List[IR.Diagnostic] = {
-    GatherDiagnostics
-      .runModule(
-        module.getIr,
-        ModuleContext(module, compilerConfig = config)
-      )
-      .unsafeGetMetadata(
-        GatherDiagnostics,
-        "No diagnostics metadata right after the gathering pass."
-      )
-      .diagnostics
+    if (module.getIr == null) List()
```

Comment:
Rather than writing `module.getIr == null` all over the place, can we have `hasIr()`, or sth similar, instead?